File: SWA/day1/Task_2/CBO/HighCBO.java
class ScreeningSchedule {
    void getShowTimes() { System.out.println("Fetching showtimes..."); }
}

class SeatMap {
    void lockSeat() { System.out.println("Seat locked."); }
}

class LoyaltySystem {
    void checkPoints() { System.out.println("Checking loyalty points..."); }
}

class PaymentGateway {
    void processPayment() { System.out.println("Processing payment..."); }
}

class TicketPrinter {
    void print() { System.out.println("Printing ticket..."); }
}

class BoxOfficeTerminal {
    ScreeningSchedule schedule = new ScreeningSchedule();
    SeatMap seatmap = new SeatMap();
    LoyaltySystem loyalty = new LoyaltySystem();
    PaymentGateway payment = new PaymentGateway();
    TicketPrinter printer = new TicketPrinter();

    void bookTicket() {
        schedule.getShowTimes();
        seatmap.lockSeat();
        loyalty.checkPoints();
        payment.processPayment();
        printer.print();
        System.out.println("Ticket booking completed!");
    }
}

public class HighCBO {
    public static void main(String[] args) {
        BoxOfficeTerminal terminal = new BoxOfficeTerminal();
        terminal.bookTicket();
        System.out.println("High CBO = 5 dependencies");
    }
}

File: SWA/day1/Task_2/CBO/highcbo.ts
class ScreeningSchedule {
  getShowTimes() { console.log("Fetching showtimes..."); }
}

class SeatMap {
  lockSeat() { console.log("Seat locked."); }
}

class LoyaltySystem {
  checkPoints() { console.log("Checking loyalty points..."); }
}

class PaymentGateway {
  processPayment() { console.log("Processing payment..."); }
}

class TicketPrinter {
  print() { console.log("Printing ticket..."); }
}

class BoxOfficeTerminal {
  schedule = new ScreeningSchedule();
  seatmap = new SeatMap();
  loyalty = new LoyaltySystem();
  payment = new PaymentGateway();
  printer = new TicketPrinter();

  bookTicket() {
    this.schedule.getShowTimes();
    this.seatmap.lockSeat();
    this.loyalty.checkPoints();
    this.payment.processPayment();
    this.printer.print();
    console.log("Ticket booking completed!");
  }
}

export function highcbo() {
  const terminal = new BoxOfficeTerminal();
  terminal.bookTicket();
  console.log("High CBO = 5 dependencies");
}

highcbo();
